client-implementation.md

```diff
@@ -0,0 +1,67 @@
+# Client Implementations
+
+This document describes client implementation recommendations for interoperability with registry servers and other clients using the OCI Distribution Spec.
+
+## Backwards Compatibility
+
+Client implementations SHOULD support registries that implement partial or older versions of the OCI Distribution Spec.
+This section describes how client fallback procedures when an API is not available.
+
+### Referrers API
+
+The Referrers API here is described by [Listing Referrers](spec.md#listing-referrers) and [end-12a](spec.md#endpoints).
+
+A client that pushes an Image or Artifact manifest with a defined `Refers` field MUST verify the Referrers API is available.
+A client querying the Referrers API and receiving a 404 MUST fallback to using an Index pushed to a tag described by the following schema.
+
+#### Referrers Tag Schema
+
+```text
+<alg>-<ref>
+```
+
+- `<alg>`: the digest algorithm (e.g. `sha256` or `sha512`)
+- `<ref>`: the digest from the `refers` field (limit of 64 characters)
+
+For example, a manifest with the `Refers` field digest set to `sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa` in the `registry.example.org/project` repository would have a expect an Index at `registry.example.org/project:sha256-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa`
+
+#### Pushing Manifests with Refers
+
+When pushing an Image or Artifact manifest with the `Refers` field and the Referrers API returns a 404, the client MUST:
```

Comment:
```suggestion
When pushing a manifest with a `Refers` field and the Referrers API is not present in the Registry (has returned a 404 on use of the `GET` Referrers endpoint), the client MUST:
```

Reply:
Setting the `Refers` field on an Index manifest is not supported (it could result in loops in the DAG that we want to avoid). So the "Image or Artifact manifest" language here is important.

Reply:
nod.. language could include or say manifest of type image/artifact.. note: a refers field was not added to image.index and will not to avoid loops in the directed graph of an image and artifact parts..